### DISCUSSION TITLE
Lesson 10: Theorical question about the LendingPoolAddressesProvider

### DISCUSSION THREAD

**Discussion by carlitox477** [smartcontractkit/full-blockchain-solidity-course-py#318] [Q&A]
I was on the part where we need to code the **get_lending_pool** function. Here, of course we need to get our **LendingPool** address to interact with the contract on the testnet or mainnetfork, but instead of getting it directly by setting the actual contract address Patrick and aave documentation suggest to get this address from the **LendingPoolAddressesProvider** because the **LendingPool** address may change (at least this is what I undestood). However, why will the **LendingPool** address change?

PD: "change" may not be accurate (contracts can't change), be useless maybe is more accurate

**Comment by cromewar** [comment]
Hello @carlitox477, if you take a deeper look into aave you will see there are different versions of the protocol they use, like `aave v1` and `aave v2`, so on those upgrades the lending pool address will change, also you are right contracts can't change, but you can upgrade them as you'll see on lesson 12, so the address could change at upgrading. This is not something that is going to happen often but is worth to take it in consideration.

In a personal note I prefer using providers as they are more decentralized an secure.

Best regards.
